### PR TITLE
Feature/new vault container

### PR DESCRIPTION
**Pull request by gtcno**
#97 Was not properly fixed.  This PR, is the second attempt of creating only one container when requesting multiple secrets from Vault. 

This PR is dependent on https://github.com/navikt/nais-yaml/issues/55

- Use new sidecar/initcontainer: https://github.com/navikt/vault-sidekick
- Only create one sidecar/initContainer.

Tests cases covers the following usecases in nais.yaml:
```
vault:
  enabled: true
```
Run as initContainer and fetch secrets from a default path( kv/cluster/env/app/namespace) to
a default mount(/var/run/secrets/nais.io/vault/)
Save vault token to /var/secrets/nais.io/vault/vault_token

```
vault:
  enabled: true
  sidecar: true
```
First run an initContainer  and fetch secrets from a default path( kv/cluster/env/app/namespace) to
a default mount(/var/run/secrets/nais.io/vault/)
Save vault token to /var/secrets/nais.io/vault/vault_token
**Refreshes the Vault token indefinitely** using a sidecar

 ```
vault:
    enabled: true
    paths:
    - mountPath: /secrets/certificate/hubba
       kvPath: /secrets/certificate/hubba
    - mountPath: /secrets/certificate/bubba
       kvPath: /secrets/certificate/bubba
```
Run as initContainer and fetch secrets from the given kvPaths and saving them to the given mounthPaths. 
Save vault token to /var/secrets/nais.io/vault/vault_token

 ```
vault:
  enabled: true
  sidecar: true  
  paths:
  - mountPath: /secrets/certificate/hubba
     kvPath: /secrets/certificate/hubba
  - mountPath: /secrets/certificate/bubba
     kvPath: /secrets/certificate/bubba
```
First run  an initContainer and fetch secrets from the given kvPaths and saving them to the given mounthPaths. 
Save vault token to /var/secrets/nais.io/vault/vault_token
**Refreshes the Vault token indefinitely** using a sidecar

### CODE REVIEW

File: pkg/vault/vaultcontainer.go [hunk 1]
```diff
@@ -68,114 +64,151 @@ func Enabled() bool {
...
+
+func (c config) createVaultContainerArgs(paths []nais.SecretPath) []string {
+
+	args := []string{
+		"-v=10",
+		"-logtostderr",
+		"-renew-token",
+		fmt.Sprintf("-vault=%s", c.vaultAddr),
+		fmt.Sprintf("-save-token=%s", defaultVaultTokenFileName()),
+	}
```
Comment (Kyrremann) [inline pkg/vault/vaultcontainer.go]: Hvor finner jeg beskrivelse på disse argumentene?

File: pkg/naiserator/config/config.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ const (
 	SecurelogsConfigMapReloadImage      = "securelogs.configmap-reload-image"
 	SecurelogsFluentdImage              = "securelogs.fluentd-image"
 	VaultAddress                        = "vault.address"
-	VaultAuthPath                       = "vault.auth-path"
+	VaultAuthPath                       = "vault.auth-path-new"
```
Comment (Kyrremann) [inline pkg/naiserator/config/config.go]: Kunne vi brukt `v2` i stedet for `new`. For hva gjør vi hvis vi ønsker å endre denne igjen?

File: pkg/naiserator/config/config.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ const (
 	SecurelogsConfigMapReloadImage = "securelogs.configmap-reload-image"
 	SecurelogsFluentdImage         = "securelogs.fluentd-image"
 	VaultAddress                   = "vault.address"
-	VaultAuthPath                  = "vault.auth-path"
+	VaultAuthPath                  = "vault.auth-path-new"
```
[the remark this answers is not in the document]
Comment (gtcno) [reply pkg/naiserator/config/config.go]: Dette er en temp commit for å sikre at jeg ikke knekker noe når denne blir PRen blir merget inn.
For slik jeg forstår naiserator pipelinen så vil den gjøre en autocommit til nais-yaml repoet som deployer til klustrerne. nais-yaml repoet inneholder konfigurasjon til naiserator. 
Så tanken var å kjøre med et dobbelt sett av env variabler til ting er stabilt så bare fjerne "new"

File: pkg/naiserator/config/config.go (hunk 2 above)
Comment (gtcno) [reply pkg/naiserator/config/config.go]: Se forrige svar.

File: pkg/vault/vaultcontainer.go (hunk 1 above)
Comment (gtcno) [reply pkg/vault/vaultcontainer.go]: https://github.com/navikt/vault-sidekick

File: pkg/naiserator/config/config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Kyrremann) [reply pkg/naiserator/config/config.go]: Ah, høres bra ut

**Comment by gtcno** [comment]
@eirslett Pointed out that if running as a sidecar we might run into timing issues where the sidecar has finished fetching the secrets which the main container are dependent on. 

Will rewrite code to handle these edge cases.

**Pull request merged**